torchtitan/config_manager.py

```diff
@@ -299,11 +299,10 @@ def __init__(self):
         self.parser.add_argument(
             "--experimental.pipeline_parallel_schedule",
             type=str,
-            choices=["1F1B", "GPipe", "Interleaved1F1B", "FlexibleInterleaved1F1B"],
             default="1F1B",
             help="""
-                Specify the Pipeline Parallel schedule to use.
-
+                Specify the Pipeline Parallel schedule to use. The supported schedules are:
+                https://github.com/pytorch/pytorch/blob/de4c2a3b4e89d96334dc678d1c3f2ae51a6630a0/torch/distributed/pipelining/schedules.py#L2161.
```

Comment:
Other than 1F1B, GPipe, etc., I also see `PipelineScheduleSingle` and `PipelineScheduleMulti` as valid inputs to `get_schedule_class`. I thought there are just base classes to inherit -- do they provide more granular control over the schedule if the user wants?

In other words, would torchtitan job succeed if we specify these two as `pipeline_parallel_schedule`? (If not we should clarify what's the accepted schedules.)

Reply:
Originally those base classes were added if the user wants to create their own schedule. We don't have the schedule load via CSV capabilities in torchtitan so i will add an error message for those schedules in particular